server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3767,4 +3610,242 @@ synchronized boolean assertConsistent() {
             return true;
         }
     }
+
+    private static final class CreateSnapshotTask implements ClusterStateTaskListener {
+
+        private final Repository repository;
+
+        private final RepositoryData repositoryData;
+
+        private final ActionListener<Snapshot> listener;
+
+        private final Snapshot snapshot;
+
+        private final CreateSnapshotRequest createSnapshotRequest;
+
+        private final RepositoryMetadata repositoryMetadata;
```

Comment:
why

all

the

spaces

? :grin:

Also it looks like this could be a `record`.

Reply:
Right lets do a record :)